### PR TITLE
Update backup api from 2019-05-13 to 2021-07-01;  allows Zone Redundant storage for azurerm_recovery_services_vault. storage_mode_type

### PR DESCRIPTION
**Pull request by josh-barker**
This PR:
- updates the backup api to 2021-07-01
- add zone redundant as allowed storage mode type in rsv

It might be easier to review by commit.
Also happy to split into 2 x PRs if you'd prefer.

Cheers

### CODE REVIEW

File: internal/services/recoveryservices/recovery_services_vault_resource.go [hunk 1]
```diff
@@ -247,8 +248,37 @@ func resourceRecoveryServicesVaultCreateUpdate(d *pluginsdk.ResourceData, meta i
...
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	// storage type is not updated instantaneously, so we wait until storage type is correct
+	err = pluginsdk.Retry(stateConf.Timeout, func() *pluginsdk.RetryError {
+		if resp, err := storageCfgsClient.Get(ctx, id.Name, id.ResourceGroup); err == nil {
+			if resp.Properties.StorageType != storageCfg.Properties.StorageModelType {
```
Comment (tombuildsstuff) [inline internal/services/recoveryservices/recovery_services_vault_resource.go:271]: it's possible for `resp.Properties` to be nil here:

```suggestion
			if resp.Properties == nil {
				return pluginsdk.NonRetryableError(fmt.Errorf("updating %s Storage Config: `properties` was nil"))
			}
			if resp.Properties.StorageType != storageCfg.Properties.StorageModelType {
```

File: internal/services/recoveryservices/recovery_services_vault_resource.go [hunk 2]
```diff
@@ -247,8 +248,37 @@ func resourceRecoveryServicesVaultCreateUpdate(d *pluginsdk.ResourceData, meta i
...
+				return pluginsdk.RetryableError(fmt.Errorf("updating Recovery Service Storage Cfg %s: %+v", id.String(), err))
```
Comment (tombuildsstuff) [inline internal/services/recoveryservices/recovery_services_vault_resource.go:272]: the `id` is wrapped below, so we can remove this:

```suggestion
				return pluginsdk.RetryableError(fmt.Errorf("updating Storage Config: %+v", err))
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @josh-barker 

Thanks for this PR - I've taken a look through and left some comments inline but this is looking good, if we can fix those up then we should be able to take another look, run the tests and get this merged 👍 

Thanks!

**Comment by josh-barker** [comment]
Hey @tombuildsstuff, no worries. I've addressed your feedback.

**Review by katbyte — APPROVED** [review]
Thanks @josh-barker - LGTM 🌳

**Pull request merged**